### DISCUSSION TITLE
"unifi-os shell" command not found

### DISCUSSION THREAD

**Discussion by mjaanes** [fabianishere/udm-iptv#187] [Q&A]
First of all: Thank you so much for the great work you have done to allow the UDM to support IPTV. I can not fathom why Unifi does not have support for this built-in.

I am a newbie and tried to install, but stopped at the very first command. 
I log successfully into my UDM Pro SE (FW 2.5.11) with root@192.168.1.1, but when trying to execute the command "unifi-os shell", I get the following error message.

-bash: unifi-os: command not found

Please help.....

Thanks.

**Comment by eradicationx** [comment]
no need to type that on the UDM SE :)

**Reply by mjaanes** [reply]
Thx. I will give it a try.
:-)

**Comment by mjaanes** [comment]
OK, so I got a step further thanks to @eradicationx :-)

 However, during installation and upon selecting TELENOR (as my IP TV provider) and pressing OK, I get an error message as follows:

/tmp/udm-iptv.config.kbAcih: 134: .: Can't open /usr/lib/udm-iptv/profiles/telenor.profile
udm-iptv failed to preconfigure, with exit status 2
Selecting previously unselected package igmpproxy.

I have check that the file telenor.profile is indeed where it is expected to be.

I have tried IP-TV on the new VLAN 90 port. TV does play initially, but freezes with an error message from the IPTV-box after 15 minutes. That kind of tells me that things are not working as they should.

Ran the "Debugging DHCP" commands. They did not seem OK.

root@MJ-UDM-SE:~# ip -4 addr show dev iptv
Device "iptv" does not exist.

root@MJ-UDM-SE:~# ip route list
85.166.28.0/22 dev eth8 proto kernel scope link src 85.166.XX.XXX
192.168.0.0/24 dev br90 proto kernel scope link src 192.168.0.1
192.168.1.0/24 dev br0 proto kernel scope link src 192.168.1.1
192.168.10.0/24 dev br99 proto kernel scope link src 192.168.10.1
root@MJ-UDM-SE:~#



Any good suggestions?
How can I check if the settings made by the program have actually been set? 

Thanks in advance




======================= START udm-iptv-diag =========================

root@MJ-UDM-SE:~# udm-iptv-diag
Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth8
WAN VLAN: 0 (dev iptv)
WAN DHCP: false (options "-O staticroutes -V IPTV_RG")
WAN Ranges: 224.0.0.0/4  93.91.111.0/24  148.122.7.125
LAN Interfaces: br0  br90
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: false
=== IP Link and Route ===
11: eth8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    inet 85.166.28.202/22 scope global dynamic eth8
       valid_lft 1086sec preferred_lft 1086sec
85.166.28.0/22 proto kernel scope link src 85.166.28.202
=== Service Logs ===
Sep 25 18:52:42 MJ-UDM-SE systemd[1]: udm-iptv.service: Consumed 91ms CPU time.
Sep 25 18:52:47 MJ-UDM-SE systemd[1]: udm-iptv.service: Service RestartSec=5s expired, scheduling restart.
Sep 25 18:52:47 MJ-UDM-SE systemd[1]: udm-iptv.service: Scheduled restart job, restart counter is at 1.
Sep 25 18:52:47 MJ-UDM-SE systemd[1]: Stopped IPTV support for the UniFi Dream Machine.
Sep 25 18:52:47 MJ-UDM-SE systemd[1]: udm-iptv.service: Consumed 0 CPU time.
Sep 25 18:52:47 MJ-UDM-SE systemd[1]: Started IPTV support for the UniFi Dream Machine.
Sep 25 18:52:47 MJ-UDM-SE udm-iptv[2719]: NATing IPTV network ranges (if necessary)
Sep 25 18:52:47 MJ-UDM-SE udm-iptv[2719]: Monitoring interface eth8 for changes
Sep 25 18:52:47 MJ-UDM-SE udm-iptv[2719]: Setting up igmpproxy
Sep 25 18:52:47 MJ-UDM-SE udm-iptv[2719]: Starting igmpproxy

**Reply by fabianishere** [reply]
Have a look at this comment: https://github.com/fabianishere/udm-iptv/discussions/59#discussioncomment-2116625

**Comment by mjaanes** [comment]
Thanks @fabianishere. 
I got things to work. I had the firewall rules mentioned above already in place. The problem seems to be that my US-24 switch floods with traffic (and IP TV stops after a few minutes), indicating that IGMP Snooping is not working on the US-24.

What I did (see attached picture) was to connect the IP-TV box directly to one of the LAN ports on the UDM PRO and on a separate VLAN 90. In addition, I connected the US-24 (rest of things) to another UDM PRO LAN port where I do NOT allow VLAN 90 to travel.
Now it seems stable (crossing fingers).

![Topology](https://user-images.githubusercontent.com/82218446/192719390-b6396db2-4b7c-4d62-a9c5-b70a33e26ba2.jpg)
